modules/machineset-creating-azure-ultra-disk.adoc

```diff
@@ -0,0 +1,291 @@
+--from-file=userData=userData.txt \
+--from-file=disableTemplating=disableTemplating.txt
+----
++
+where #need to understand what these values are and where they come from#
+endif::mapi[]
+
+. Edit or copy an existing Azure `MachineSet` custom resource (CR) by running the following command:
++
+[source,terminal]
+----
+$ oc edit machineset <machine-set-name>
+----
++
+where `<machine-set-name>` is the machine set that you want to provision machines on ultra disks.
+
+. Add the following lines in the positions indicated:
++
+[source,yaml]
+----
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+  ...
+spec:
+  ...
+  template:
+    ...
+    spec:
+      metadata:
+        ...
+        labels:
+          ...
+          disk: ultrassd <1>
+          ...
+      providerSpec:
+        value:
+          ...
+          ultraSSDCapability: Enabled <1>
+ifdef::mapi[]
+          dataDisks: <1>
+          - nameSuffix: ultrassd
+            lun: 0
+            diskSizeGB: 4
+            deletionPolicy: Delete
+            cachingType: None
+            managedDisk:
+              storageAccountType: UltraSSD_LRS
+endif::mapi[]
+             ...
+----
++
+<1> These lines enable the use of ultra disks.
+ifdef::mapi[For `dataDisks`, include the entire stanza.]
+
+. Create a machine set using the updated configuration by running the following command:
++
+[source,terminal]
+----
+$ oc create -f <machine-set-name>.yaml
+----
+
+ifdef::pvc[]
+. Create a storage class that contains the following YAML definition:
++
+[source,yaml]
+----
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  creationTimestamp: "2022-04-11T11:02:53Z"
+  name: ultra-disk-sc <1>
+  resourceVersion: "56877"
+  uid: 08b2d2ca-adbf-4f6e-b333-1612daf374ad
+parameters:
+  cachingMode: None
+  diskIopsReadWrite: "2000"
+  diskMbpsReadWrite: "320"
+  kind: managed
+  skuname: UltraSSD_LRS
+provisioner: disk.csi.azure.com <2>
+reclaimPolicy: Delete
+volumeBindingMode: WaitForFirstConsumer <3>
+----
+<1> Specify the name of the storage class. This procedure uses `ultra-disk-sc` for this value.
+<2> For Azure Kubernetes Service (AKS) version 1.21 or later, use `disk.csi.azure.com`. For earlier versions of AKS, use `kubernetes.io/azure-disk`.
+<3> Optional: Specify this parameter to wait for the creation of the pod that will use the disk.
+#Which of the above lines are required? Which lines have to say something specific vs. being the user's choice?#
+
+. Create a persistent volume claim (PVC) to reference the `ultra-disk-sc` storage class that contains the following YAML definition:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: ultra-disk
+spec:
+  accessModes:
+  - ReadWriteOnce
+  storageClassName: ultra-disk-sc
+  resources:
+    requests:
+      storage: 4Gi
+----
+#Which of the above lines are required? Which lines have to say something specific vs. being the user's choice?#
+
```

Comment:
The Below is validation for our settings to use ultraSSD , we have multiple ways an admin can confirm this , from GUI also he can check . May be you can leave this one as example .
The moment disk is attached to the machine and workload running on the node created by the machine is not showing any errors , we can ascertain the ultraSSD storage disk is working fine with the machine-api.

All the comments that I have provided are based on - https://github.com/openshift/enhancements/blob/master/enhancements/machine-api/azure-ultra-disks.md#implementation-detailsnotesconstraints 

Very helpful guide created by @damdo , may be if we can insert its reference as well , it might help .

Reply:
Haha sorry the "holy heck rewrite this" was aimed at myself. I just made a big long sentence as a placeholder and hadn't broken it up yet :sweat_smile: Thanks for the info in your comment for this!